### PR TITLE
[Ide] Sets default naming and adds extension checks when saving

### PR DESCRIPTION
**Pull request by jtorres**
- Adds default tab naming
- Additional checks when saving .binlog file

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/BuildOutputWidget.cs [hunk 1]
```diff
@@ -120,10 +121,15 @@ void Initialize ()
...
 			saveButton.Clicked += async (sender, e) => {
 				var dlg = new OpenFileDialog (GettextCatalog.GetString ("Save as..."), MonoDevelop.Components.FileChooserAction.Save) {
-					TransientFor = IdeApp.Workbench.RootWindow
+					TransientFor = IdeApp.Workbench.RootWindow,
+					InitialFileName = string.IsNullOrEmpty (ViewContentName) ? editor.FileName.FileName : ViewContentName
 				};
 				if (dlg.Run ()) {
-					await BuildOutput.Save (dlg.SelectedFile);
+					var outputFile = dlg.SelectedFile;
+					if (!outputFile.HasExtension ("binlog"))
```
Comment (netonjm) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/BuildOutputWidget.cs]: binlog must be a constant

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/BuildOutputWidget.cs [hunk 2]
```diff
@@ -120,10 +121,15 @@ void Initialize ()
 			saveButton.Accessible.SetTitle (saveLbl.Text);
 			saveButton.Clicked += async (sender, e) => {
 				var dlg = new OpenFileDialog (GettextCatalog.GetString ("Save as..."), MonoDevelop.Components.FileChooserAction.Save) {
-					TransientFor = IdeApp.Workbench.RootWindow
+					TransientFor = IdeApp.Workbench.RootWindow,
+					InitialFileName = string.IsNullOrEmpty (ViewContentName) ? editor.FileName.FileName : ViewContentName
```
Comment (netonjm) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/BuildOutputWidget.cs]: ViewContentName ?? editor.FileName.FileName

**Comment by rodrmoya** [comment]
Code looks good to me, but when saving the build output, the tab name doesn't change to the new file, so seems the `BuildOutputWidget` needs to notify the `BuildOutputViewContent` about the file name change?

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/BuildOutputWidget.cs (hunk 2 above)
Comment (jtorres) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/BuildOutputWidget.cs]: the most likely is that to be empty, so I want to check both, or empty or null

**Pull request merged**